intfconfd/intfconf.cpp

```diff
@@ -0,0 +1,120 @@
+#include <string.h>
+#include "logger.h"
+#include "dbconnector.h"
+#include "producerstatetable.h"
+#include "tokenize.h"
+#include "ipprefix.h"
+#include "intfconf.h"
+#include "exec.h"
+
+using namespace std;
+using namespace swss;
+
+#define VLAN_PREFIX         "Vlan"
+#define LAG_PREFIX          "PortChannel"
+
+IntfConf::IntfConf(DBConnector *cfgDb, DBConnector *appDb, DBConnector *stateDb, vector<string> tableNames) :
+        CfgOrch(cfgDb, tableNames),
+        m_cfgIntfTable(cfgDb, CFG_INTF_TABLE_NAME, CONFIGDB_TABLE_NAME_SEPARATOR),
+        m_cfgVlanIntfTable(cfgDb, CFG_VLAN_INTF_TABLE_NAME, CONFIGDB_TABLE_NAME_SEPARATOR),
+        m_statePortTable(stateDb, STATE_PORT_TABLE_NAME, CONFIGDB_TABLE_NAME_SEPARATOR),
+        m_stateLagTable(stateDb, STATE_LAG_TABLE_NAME, CONFIGDB_TABLE_NAME_SEPARATOR),
+        m_stateVlanTable(stateDb, STATE_VLAN_TABLE_NAME, CONFIGDB_TABLE_NAME_SEPARATOR),
+        m_appIntfTableProducer(appDb, APP_INTF_TABLE_NAME)
+{
+
+}
+
+void IntfConf::syncCfgDB()
+{
+    CfgOrch::syncCfgDB(CFG_INTF_TABLE_NAME, m_cfgIntfTable);
+    CfgOrch::syncCfgDB(CFG_VLAN_INTF_TABLE_NAME, m_cfgVlanIntfTable);
+}
+
+bool IntfConf::setIntfIp(string &alias, string &opCmd, string &ipPrefixStr)
+{
+    string cmd, res;
+
+    cmd = "ip address " + opCmd + " ";
+    cmd += ipPrefixStr + " dev " + alias;
+    swss::exec(cmd, res);
+    return true;
+}
+
+bool IntfConf::isIntfStateOk(string &alias)
```

Comment:
>isIntfStateOk [](start = 15, length = 13)

suggest name: hasIntf

Reply:
Honestly I don't see readyIntf a better name than isIntfStateOk,  we do put state ok in stateDB though it is checked  for now.  If you insist on changing the name, I could do it.

Reply:
Just checking the code, there is nothing about "Ok" state. So I guess the original name is not perfect.

Reply:
Ok,  will add code to check the state:

127.0.0.1:6379[6]> hgetall "PORT_TABLE|Ethernet47"
1) "state"
2) "ok"

Reply:
I don't see that you check "ok"